packages/main/src/plugin/cli-tool-registry.ts

```diff
@@ -82,17 +89,52 @@ export class CliToolRegistry {
     this.apiSender.send('cli-tool-remove', cliTool.id);
   }
 
-  getCliToolInfos(): CliToolInfo[] {
-    return Array.from(this.cliTools.values()).map(cliTool => {
-      return {
+  async getCliToolInfos(): Promise<CliToolInfo[]> {
+    const cliToolInfos: CliToolInfo[] = [];
+    for (const cliTool of this.cliTools.values()) {
+      const cliVersion = await cliTool.getVersion();
+      cliToolInfos.push({
```

Comment:
There should be not so many calls go get all CliToolInfo's. Only at start or if manual reload requested (not if it is going to be needed). Mostly events are going to be sent out about added/removed/changed items.